atom/renderer/api/atom_api_renderer_ipc.cc

```diff
@@ -109,32 +110,50 @@ class IPCRenderer : public mate::Wrappable<IPCRenderer> {
   }
 
  private:
-  static void SendMessageSyncOnWorkerThread(
+  void SendMessageSyncOnWorkerThread(
       atom::mojom::ElectronBrowserPtrInfo* interface_info,
       base::WaitableEvent* event,
       base::Value* result,
       bool internal,
       const std::string& channel,
       const base::ListValue* arguments) {
     atom::mojom::ElectronBrowserPtr browser_ptr(std::move(*interface_info));
-    browser_ptr->MessageSync(
+    worker_thread_electron_browser_ptr_ = std::move(browser_ptr);
+
+    worker_thread_electron_browser_ptr_.set_connection_error_handler(
+        base::BindOnce(&IPCRenderer::HandleMojoConnectionErrorOnWorkerThread,
+                       base::Unretained(&worker_thread_electron_browser_ptr_),
```

Comment:
it might be simpler to pass `base::Unretained(this)` and have the method be non-static? that way we don't have to pass all these member variables around

Reply:
Yep, could do; was just thinking I'd match the style of the surrounding code, which passes things around. It of course changes in 7-0-x, so this might not be worth debating.